### PR TITLE
#700 Fixing enum constants and adding unit test.

### PR DESCRIPTION
**Pull request by sjaakd**
There's still #871 as small spin of to be taken care of..

### CODE REVIEW

File: processor/src/main/java/org/mapstruct/ap/internal/model/assignment/EnumConstantWrapper.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    private final Type enumType;
+
+    public EnumConstantWrapper(Assignment decoratedAssignment, Type enumType ) {
+        super( decoratedAssignment );
+        this.enumType = enumType;
+    }
+
+    @Override
+    public Set<Type> getImportTypes() {
+        Set<Type> imported = new HashSet<Type>( getAssignment().getImportTypes().size() + 1 );
```
Comment (agudian) [inline processor/src/main/java/org/mapstruct/ap/internal/model/assignment/EnumConstantWrapper.java]: That would require a resize of the underlying map, as the default load factor is 0.75... Let's just use `new HashSet<Type>( getAssignment().getImportTypes() )` followed by just the `.add( enumType )`.

**Comment by agudian** [comment]
Cool thing! I only have a teeny tiny remark. Otherwise feel free to push!

**Comment by sjaakd** [comment]
test this please

**Comment by sjaakd** [comment]
test this please

**Pull request merged**